clang_build/project.py

```diff
@@ -0,0 +1,300 @@
+
+        targets_and_subproject_targets = self.targets_config.copy()
+        for project in subprojects:
+            targets_and_subproject_targets.update(project.targets_config)
+
+        # Parse targets from toml file
+        # print(targets_and_subprojects)
+        non_existent_dependencies = _find_non_existent_dependencies(targets_and_subprojects)
+        if non_existent_dependencies:
+            error_messages = [f'In {target}: the dependency {dependency} does not point to a valid target' for\
+                            target, dependency in non_existent_dependencies]
+
+            error_message = _textwrap.indent('\n'.join(error_messages), prefix=' '*3)
+            environment.logger.error(error_message)
+            # print(f"Project {self.name}: non_existent_dependencies.")
+            # print(f"                     ", error_messages)
+            sys.exit(1)
+
+        circular_dependencies = _find_circular_dependencies(self.targets_config)
+        if circular_dependencies:
+            error_messages = [f'In {target}: circular dependency -> {dependency}' for\
+                            target, dependency in non_existent_dependencies]
+
+            error_message = _textwrap.indent('\n'.join(error_messages), prefix=' '*3)
+            environment.logger.error(error_message)
+            print(f"Project {self.name}: circular_dependencies.")
+            sys.exit(1)
+
+
+        target_names_total = _get_dependency_walk(targets_and_subproject_targets)
+        target_names_project = []
+        for name in target_names_total:
+            if name in self.targets_config:
+                target_names_project.append(name)
+        # print(f"-- Project {self.name}: target names of project and subprojects: ", target_names_total)
+        # print(f"-- Project {self.name}: target names of project: ", target_names_project)
+        # print(f"-- Project {self.name}: target config: ", targets_and_subproject_targets)
+
+        self.target_list = []
+
+
+        for target_name in _IteratorProgress(target_names_project, environment.progress_disabled, len(target_names_project)):
+            # print(f"-- Project {self.name}: target {name}:................")
+            target_node = targets_and_subproject_targets[target_name]
+            # Directories
+            target_build_dir = self.build_directory if not multiple_targets else self.build_directory.joinpath(target_name)
+            # Sources
+            files = _get_sources_and_headers(target_node, self.workingdir, target_build_dir)
+            # Dependencies
+            dependencies = []
+            for name in target_node.get('dependencies', []):
+                subnames = name.split(".")
+                if subnames[0] in self.subproject_names:
+                    idx = self.subproject_names.index(subnames[0])
+                    # TODO: so far, we are only going one layer deep... this is not enough
```

Comment:
I guess this still needs to be figured out.

Reply:
I would implement this properly in a follow-up pull request, IMO the feature works well enough for now.